tl

```diff
@@ -50,36 +59,33 @@ end
 package.path = script_path() .. "/?.lua;" .. package.path
 
 local tl = require("tl")
+local argparse = require("argparse")
 
-local tlconfig = get_config()
+local parser = argparse("tl", "A minimalistic typed dialect of Lua.")
 
-local cmd
-local filename
-
-local is_cmd = {
-   run = true,
-   check = true,
-   gen = true,
-}
-
-if is_cmd[arg[1]] then
-   cmd = arg[1]
-   filename = arg[2]
-else
-   cmd = "check"
```

Comment:
Is there a way in argparse to keep supporting `tl file.tl` without an explicit command?

To be honest, I don't know if it's just my habit of typing `tl tl.tl` speaking or if it's really something worth keeping because it's probably the first thing a user would try...

Reply:
Mmmh... It's kinda hacky, but we could call
```lua
local args, err = parser:pparse()
```
...and check if `err` starts with `unknown command`. If it does, we could assume that the 'unknown command' is actually a file.

Maybe there's a better way of doing this...

Reply:
See https://github.com/hishamhm/tl/pull/42/commits/32b777e32ddde7af051d434cd5f861caaf33c42d.

Reply:
Aw, yeah, that _is_ pretty hacky. And mixing argument 1 with commands and filenames is conceptually ugly, the more I think about it.

Reply:
I think we could drop that and require `tl check file.tl` indeed.